File: Data_Structures/python/01_음계.py
list_temp = list(map(int, input().split()))

result = False

for i in range(0, 8):
    if list_temp[i] != i + 1:
        result = True
        break
    elif i+1 == 8:
        print("ascending")

if result == True:
    for i in range(0, 8):
        if list_temp[i] != 8-i:
            result = True
            break
        elif i+1 == 8:
            print("descending")
            result = False

if result == True:
    print("mixed")

File: Data_Structures/c++/01_음계.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/2920
// 2

#include <iostream>

using namespace std;

int main()
{
	int n[8] = { 0, };
	for (int i = 0; i < 8; i++)
		cin >> n[i];

	bool ascending = true, descending = true;
	for (int i = 1; i < 8; i++) {
		if (n[i] < n[i - 1])
			ascending = false;
		else if (n[i] > n[i - 1])
			descending = false;
	}

	if (ascending)
		cout << "ascending" << endl;
	else if (descending)
		cout << "descending" << endl;
	else cout << "mixed" << endl;
}